tv_apps/smart-system/js/focus_manager.js

```diff
@@ -29,29 +29,41 @@
   };
 
   proto.focus = function ff_focus() {
-    // We should get the object reference when we try to focus because smart
-    // system has a lot of lazy load mechanism and we may get undefined when
-    // they aren't lazy loaded.
-    var systemUIList = [
-      TrustedUIManager,
-      sleepMenu
-    ];
-    // list all visible system UI
-    var visible = systemUIList.filter(function(item) {
-      return item && item.isVisible();
-    });
-    if (visible.length > 0) {
-      // sort system UIs
-      visible.sort(function(item1, item2) {
-        return item2.getOrder() - item1.getOrder();
+    // According to gecko's implementation, we cannot change focus while
+    // handling keyboard event. The way to bypass it is use setTimeout to lazy
+    // run the focus switching.
+    //
+    // There is another focus rule at gecko: we cannot move focus cross domain,
+    // like to move focus from a button is remote iframe to system app's button.
```

Comment:
There is another focusing rule in gecko: we cannot move focus across domains, like moving focus from a button in a remote iframe to another one in system app.